src/huggingface_hub/hub_mixin.py

```diff
@@ -302,3 +378,43 @@ def push_to_hub(
         repo.git_add()
         repo.git_commit(commit_message)
         return repo.git_push()
+
+
+class ModelHubMixin(BaseHfHubMixin):
```

Comment:
This class should be named `PyTorchModelHubMixin` or `PyTorchModuleHubMixin`. Leaving as `ModelHubMixin` to avoid breaking things for libraries already integrated with hub.

If you agree, we should have some deprecation strategy here